From: =?utf-8?q?Lo=C3=AFc_Minier?= <lool@dooz.org>
Date: Sat, 21 Sep 2013 23:37:03 +0200
Subject: Don't query immodules for the local tree under --disable-shared

[mbiebl: follow upstream and use immodules.cache instead of gtk.immodules]

Bug: http://bugzilla.gnome.org/show_bug.cgi?id=346531
Forwarded: yes
Applied-upstream: no, bug was closed due to lack of feedback
---
 configure.ac              |  1 +
 modules/input/Makefile.am | 17 ++++++++++++++++-
 2 files changed, 17 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 44d3056..0294160 100644
--- a/configure.ac
+++ b/configure.ac
@@ -182,6 +182,7 @@ fi
 
 AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
+AM_CONDITIONAL([ENABLE_SHARED], [test "$enable_shared" = yes])
 dnl when using libtool 2.x create libtool early, because it's used in configure
 m4_ifdef([LT_OUTPUT], [LT_OUTPUT])
 
diff --git a/modules/input/Makefile.am b/modules/input/Makefile.am
index c8610fa..bfe5085 100644
--- a/modules/input/Makefile.am
+++ b/modules/input/Makefile.am
@@ -189,7 +189,11 @@ dist_imconffile_DATA = im-multipress.conf
 if CROSS_COMPILING
 RUN_QUERY_IMMODULES_TEST=false
 else
+if ENABLE_SHARED
 RUN_QUERY_IMMODULES_TEST=test -z "$(DESTDIR)"
+else
+RUN_QUERY_IMMODULES_TEST=false
+endif
 endif
 
 # Running this if cross compiling or if DESTDIR is set is going to
@@ -253,7 +257,18 @@ noinst_LTLIBRARIES =				\
 included-modules: $(noinst_LTLIBRARIES)
 
 immodules.cache: Makefile.am $(module_LTLIBRARIES)
-	$(top_builddir)/gtk/gtk-query-immodules-2.0 $(module_LTLIBRARIES) > immodules.cache
+	@if $(RUN_QUERY_IMMODULES_TEST) ; then \
+	  echo "$(top_builddir)/gtk/gtk-query-immodules-2.0 > immodules.cache" ; \
+	  $(top_builddir)/gtk/gtk-query-immodules-2.0 $(module_LTLIBRARIES) > immodules.cache; \
+	else \
+	  echo "***" ; \
+	  echo "*** Warning: immodules.cache not built" ; \
+	  echo "***" ; \
+	  echo "*** Generate this file manually on host" ; \
+	  echo "*** system using gtk-query-immodules-2.0" ; \
+	  echo "***" ; \
+	  touch immodules.cache; \
+        fi
 
 CLEANFILES = immodules.cache
 
